Automattic/woocommerce-payments
# Add subscription variation support to WooPay button

Fixes #7588

#### Changes proposed in this Pull Request

This adds support for the `subscription_variation` product type to the WooPay button, similar to [what's done for payment request buttons](https://github.com/Automattic/woocommerce-payments/blob/0640c56368ba792b20164a48e8811194b402e299/includes/class-wc-payments-payment-request-button-handler.php#L1185).









#### Testing instructions





This covers an edge case and is a bit awkward to test. Variable Subscription products have the type, `variable-subscription` and the actual variations have the type,`subscription_variation`. When navigating to a variable subscription product and purchasing a variation, the the express checkout buttons will see `variable-subscription` as the product type. There is some case in which a subscription variation can be purchased while not attached to it's parent product, at which point the product type will be `subscription_variation`. Myself and others haven't been able to find a way to purchase a subscription variation via normal core flows.

### Instructions

1. Install WooCommerce Subscriptions and create a variable subscription with a few variations
2. Find the ID of one variation in the **Variations** tab within the product settings
3. Set the `$product_id` to the ID of the variation [here](https://github.com/Automattic/woocommerce-payments/blob/0640c56368ba792b20164a48e8811194b402e299/includes/class-wc-payments-woopay-button-handler.php#L229). This will result in that `subscription_variation` being added to the cart any time the WooPay or payment request button is used.
4. Navigate to any product and purchase via the WooPay button. _You must be logged in to purchase subscriptions via WooPay_
5. Verify the subscription variation was added to the cart and can be purchased in WooPay and that the total is correct
6. Repeat steps 4 and 5 using Google Pay, paying particular attention that the total is correct and not $0.


-------------------


**Post merge**
